### PR TITLE
Refactor wasm convert.go

### PR DESCRIPTION
**Pull request by ingwonsong**
Refactoring the `pkg/wasm/convert.go` based on the review from https://github.com/istio/istio/pull/42928#discussion_r1084368557

### CODE REVIEW

**Comment by ingwonsong** [comment]
@howardjohn @kyessenov Can you review this PR?
This is a kind of refactoring of PR https://github.com/istio/istio/pull/42928

**Review by howardjohn — COMMENTED** [review]
LGTM will let Kuat take a look though

File: pkg/wasm/convert.go [hunk 1]
```diff
@@ -44,11 +45,13 @@ func createAllowAllFilter(name string) (*anypb.Any, error) {
 
 // MaybeConvertWasmExtensionConfig converts any presence of module remote download to local file.
 // It downloads the Wasm module and stores the module locally in the file system.
-func MaybeConvertWasmExtensionConfig(resources []*anypb.Any, cache Cache) bool {
+func MaybeConvertWasmExtensionConfig(resources []*anypb.Any, cache Cache) error {
 	var wg sync.WaitGroup
+
+	var convertErr atomic.Value
```
[the remark this answers is not in the document]
Comment (ingwonsong) [reply pkg/wasm/convert.go]: Can you take a look once again? I changed to use a list and multierror package.

**Comment by ingwonsong** [comment]
ARM64 infra seems to be not working.

**Review by howardjohn — APPROVED** [review]

**Review by kyessenov — APPROVED** [review]
P&T ownership.

**Pull request merged**